File: courses/POO/Curso Mello (2009)/Archives/COMP07 Files/aulas/ex09/Biblioteca.cc
#include "Biblioteca.h"

// biblioteca
void Biblioteca::init() { 
	users  = new Vector<User *>();
	livros = new Vector<Livro *>();

	users->init();
	livros->init();
}
void Biblioteca::menu() { }
void Biblioteca::auth() { }
void Biblioteca::search() { }

// livros
void Biblioteca::add() { 
	char isbn[12];
	char title[200];
	char author[100];

	cout << "Digite os dados do Livro" << endl;
	cout << "Isbn: " << endl;
	cin >> isbn;
	cout << "Titulo: " << endl;
	cin >> title;
	cout << "Autor: " << endl;
	cin >> author;

	Livro *l = new Livro();
	l->setIsbn(isbn);
	l->setTitle(title);
	l->setAuthor(author);

	livros->add(l);
}
void Biblioteca::del() { }
void Biblioteca::list() { 
	Livro *tmp = NULL;
	int i;

	for (i = 0; i < livros->size(); i++) {
		tmp = livros->get(i);
		cout << "Isbn: "   << tmp->getIsbn() << endl;
		cout << "Titulo: " << tmp->getTitle() << endl;
		cout << "Autor: "  << tmp->getAuthor() << endl;
	}
}

// users
void Biblioteca::create() { }
void Biblioteca::remove() { }
void Biblioteca::show() { }

int main(int argc, char* argv[]) {
	Biblioteca *b = new Biblioteca();

	b->init();
	b->add();
	b->add();
	b->add();
	b->list();
}

File: courses/POO/Curso Mello (2009)/Archives/COMP07 Files/aulas/ex11/Biblioteca.java
import java.util.Vector;
import java.io.*;

public class Biblioteca {

	private Vector books;
	private Vector users;

	public Biblioteca() {
		books = new Vector();
		users = new Vector();
	}

	// Book
	public void addBook() throws Exception {
		String isbn, title, author;

		BufferedReader br = 
			new BufferedReader(new InputStreamReader(System.in));

		System.out.println("Digite os dados do Livro");
		System.out.print("Isbn: ");
		isbn = br.readLine();
		System.out.print("Title: ");
		title = br.readLine();
		System.out.print("Author: ");
		author = br.readLine();
		
		Book b = new Book(isbn, title, author);

		books.add(b);
	}

	public void listBooks() {
		for (int i = 0; i < books.size(); i++) {
			Book b = (Book) books.elementAt(i);
			System.out.println(b.getIsbn()+", "+
					b.getTitle()+", "+
					b.getAuthor());
		}
	}

	public static void main(String args[]) throws Exception {
		Biblioteca b = new Biblioteca();
		b.addBook();
		b.addBook();
		b.addBook();
		b.listBooks();
	}
}
